app/src/main/java/com/example/scoutconcordia/CalendarActivity.java

```diff
@@ -42,9 +103,388 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
+        try {
+          for (int j = startRow; j < totalRows + startRow; ++j) {
+              TextView textView = (TextView) findViewById(tableIds[eventDayColumn][j]);
+              textView.setTextColor(getResources().getColor(R.color.white));
+              textView.setBackgroundColor(colorId);
+              if (j == startRow) {
+                  textView.setText(summary);
+              } else if (j == startRow + 1) {
+                  textView.setText(location);
+              } else if (j == startRow + 2) {
+                  textView.setText(start + " - " + end);
+              }
+              attachOnClickListener(textView);
+          }
+      }
+      catch (Throwable t){
+          Log.d(TAG, t.toString());
+          Log.d(TAG, "START:" +start + ", END:"+end);
+      }
+    }
+
+    private void attachOnClickListener(TextView textView){
+        textView.setClickable(true);
+        textView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                directionsDialog().show();
+            }
+        });
+    }
+
+    private Dialog directionsDialog(){
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage("Go to Location")
+                .setCancelable(true)
+                .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
+                    }
+                })
+                .setPositiveButton("GO", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                    }
+                });
+        return builder.create();
+    }
+
+    // returns the row number that the event begins at (e.g. if event start at 9, 5 will be returned, (9-8)*4 +(0/15)+1
+    // begins at 8:00 = row 1, at 8:15 = row 2, and so on
+    private int eventStartRow(Date start, Date end){
+        java.util.Calendar c = java.util.Calendar.getInstance();
+        c.setTime(start);
+
+        int startHours = c.get(java.util.Calendar.HOUR_OF_DAY);
+        int startMins = c.get(java.util.Calendar.MINUTE);
+        int startRow = (startHours - 8) * 4 + (startMins / 15) + 1;
+
+        return startRow;
+    }
+
+
+    // returns a multidimensional array that holds ids of all text views for each column in the table Layout in activity_calendar.xml for displaying schedule
+    private int[][] tableLayoutIds(){
+        int[][] tableIds = new int[7][61];  // [x][y], x = 0 is the first row from the left in the table
+        // y = 1 (0 is not assigned) is the first column (from 8:00 to 8:15)
+        Resources r = getResources();
+        String name = getPackageName();
+
+        for (int i = 1; i < 61; i++) {
```

Comment:
is i<61 because it corresponds to 60 minutes in an hour?

Reply:
![image](https://user-images.githubusercontent.com/46825370/76707695-ceaa1700-66c7-11ea-8c1a-75b28f01156c.png)

No. Every row in the table is representing a 15 min interval. School hours are from 8:00 - 23:00. A total of 60 rows are required to represent all 15 minutes intervals. Therefore 8:00 - 8:15 would be row 1, 8:15 - 8:30 would be row 2, and 10:45 - 11:00 would be row 60.